### PR TITLE
delete resource field

### PR DESCRIPTION
**Pull request by shubham-cmyk**

### CODE REVIEW

**Comment by shubham-cmyk** [comment]
@eddycharly PTAL

File: pkg/apis/v1alpha1/test_step_spec.go [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
 	Apply []Apply `json:"apply,omitempty"`
 	// +optional
 	Error []Error `json:"error,omitempty"`
+
+	// List of object that should be deleted before the test step is executed
+	// +optional
+	Delete []ObjectReference `json:"delete,omitempty"`
+}
+
+type ObjectReference struct {
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/test_step_spec.go:20]: put it in a separate file, add comments on the struct and fields.

File: pkg/apis/v1alpha1/test_step_spec.go [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+	// +optional
+	v1.ObjectReference `json:",inline"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/test_step_spec.go:24]: this should be `corev1`

File: pkg/client/delete.go [hunk 3]
```diff
@@ -2,9 +2,15 @@ package client
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	"github.com/kyverno/chainsaw/pkg/apis/v1alpha1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	kerror "k8s.io/apimachinery/pkg/api/errors"
```
Comment (eddycharly) [inline pkg/client/delete.go:10]: 🤔

Comment (shubham-cmyk) [reply pkg/client/delete.go:10]: Oh shit didn't saw that and didn't even git a lint issue

**Comment by shubham-cmyk** [comment]
@eddycharly PTAL

**Comment by eddycharly** [comment]
@shubham-cmyk conflict

**Comment by shubham-cmyk** [comment]
> @shubham-cmyk conflict

PTAL the conflict arise due to recently merged PR

**Pull request merged**